Algorithm/QCAlgorithm.Trading.cs

```diff
@@ -198,20 +198,6 @@ public OrderTicket MarketOrder(Symbol symbol, double quantity, bool asynchronous
         public OrderTicket MarketOrder(Symbol symbol, decimal quantity, bool asynchronous = false, string tag = "")
         {
             var security = Securities[symbol];
-
-            // check the exchange is open before sending a market order, if it's not open
-            // then convert it into a market on open order
-            if (!security.Exchange.ExchangeOpen)
```

Comment:
I don't think we want to remove this -- just make `SetHoldings` issue the correct order type.

Consider an algorithm with daily data. Right now submitting `MarketOrder` just works as they get converted to `MarketOnOpenOrder` calls -- if we remove this, the `MarketOrder` will fail due to market being closed (always closed w/ daily data!)